backend/main/test_utils/leaderboard_utils.go

```diff
@@ -0,0 +1,52 @@
+package test_utils
+
+func (otu *OverflowTestUtils) GenerateLeaderboardBaseCase(communityId int) {
+	proposalIds := otu.AddActiveProposals(communityId, 3)
+	voteChoice := "a"
+
+	otu.CreateVoteAPI(proposalIds[0], otu.GenerateValidVotePayload("user1", proposalIds[0], voteChoice))
+	otu.CreateVoteAPI(proposalIds[1], otu.GenerateValidVotePayload("user1", proposalIds[1], voteChoice))
+	otu.CreateVoteAPI(proposalIds[2], otu.GenerateValidVotePayload("user1", proposalIds[2], voteChoice))
+	otu.CreateVoteAPI(proposalIds[0], otu.GenerateValidVotePayload("user2", proposalIds[0], voteChoice))
+	otu.CreateVoteAPI(proposalIds[1], otu.GenerateValidVotePayload("user2", proposalIds[1], voteChoice))
+}
```

Comment:
for this function and for `GenerateLeaderboardWithEarlyVotes` you could pass in a parameter of type `proposalIds[]` iterate over it and get the values at each index to pass in to `CreateVoteApi`

Reply:
Done.